### PR TITLE
MAINT: Remove old PY_VERSION_HEX and sys.version_info code

### PR DESCRIPTION
**Pull request by hugovk**
Remove old code wrapped in `PY_VERSION_HEX` and `sys.version_info` checks for no longer supported Python versions (<3.6).

I didn't touch any compat files, as they may still be used by downstream libraries.

Helps https://github.com/numpy/numpy/issues/15306.

### CODE REVIEW

**Comment by charris** [comment]
Let's wait on this a bit, we still support 3.6 in NumPy 1.19 and keeping it for 1.20 is still under discussion.

**Comment by hugovk** [comment]
This keeps 3.6, and only removes stuff for 3.5 and lower.

File: numpy/__init__.pyi [hunk 1]
```diff
@@ -434,11 +418,6 @@ class timedelta64(signedinteger):
     @overload
     def __add__(self, other: datetime64) -> datetime64: ...
     def __sub__(self, other: Union[timedelta64, int]) -> timedelta64: ...
-    if sys.version_info[0] < 3:
-        @overload
-        def __div__(self, other: timedelta64) -> float: ...
-        @overload
-        def __div__(self, other: float) -> timedelta64: ...
     @overload
```
Comment (eric-wieser) [inline numpy/__init__.pyi:421]: I think these changes conflict with an open PR

Comment (hugovk) [reply numpy/__init__.pyi:421]: PR https://github.com/numpy/numpy/pull/17188 includes some but not all of these.

Would you prefer that's merged first?

Or remove these changes from this PR?

**Comment by charris** [comment]
Thanks @hugovk . @BvB93  Take note, this may affect #17188.

**Pull request merged**